dds/DCPS/RTPS/Sedp.h

```diff
@@ -307,9 +308,12 @@ class Sedp : public DCPS::EndpointManager<ParticipantData_t> {
     }
 #endif
 
+    void shutting_down() { shutting_down_ = true; }
+
   protected:
     DCPS::RepoId repo_id_;
     Sedp& sedp_;
+    ACE_Atomic_Op<ACE_SYNCH_MUTEX, bool> shutting_down_;
```

Comment:
I recognize you probably copied this from below... but does ACE_SYNCH_MUTEX buy us anything here over ACE_Thread_Μutex? The ability to compile when mutexes aren't supported? The ability to make this a NOOP whenever we want to make our ACE_Atomic_Op unatomic?

Reply:
Yes, it was just a refactor from derived to base.  We can change it.